packages/graphql/src/schema-model/annotation/Annotation.ts

```diff
@@ -17,140 +17,83 @@
+import { parseJWTPayloadAnnotation } from "../parser/annotations-parser/jwt-payload-annotation";
+import { parseMutationAnnotation } from "../parser/annotations-parser/mutation-annotation";
+import { parsePluralAnnotation } from "../parser/annotations-parser/plural-annotation";
+import { parsePopulatedByAnnotation } from "../parser/annotations-parser/populated-by-annotation";
+import { parsePrivateAnnotation } from "../parser/annotations-parser/private-annotation";
+import { parseQueryAnnotation } from "../parser/annotations-parser/query-annotation";
+import { parseLimitAnnotation } from "../parser/annotations-parser/limit-annotation";
+import { parseSelectableAnnotation } from "../parser/annotations-parser/selectable-annotation";
+import { parseSettableAnnotation } from "../parser/annotations-parser/settable-annotation";
+import { parseSubscriptionAnnotation } from "../parser/annotations-parser/subscription-annotation";
+import { parseSubscriptionsAuthorizationAnnotation } from "../parser/annotations-parser/subscriptions-authorization-annotation";
+import { parseTimestampAnnotation } from "../parser/annotations-parser/timestamp-annotation";
+import { parseUniqueAnnotation } from "../parser/annotations-parser/unique-annotation";
+import type { DirectiveNode } from "graphql";
+import { parseKeyAnnotation } from "../parser/annotations-parser/key-annotation";
 
-export type Annotation =
-    | CypherAnnotation
-    | AuthorizationAnnotation
-    | AuthenticationAnnotation
-    | KeyAnnotation
-    | SubscriptionsAuthorizationAnnotation
-    | LimitAnnotation
-    | DefaultAnnotation
-    | CoalesceAnnotation
-    | CustomResolverAnnotation
-    | IDAnnotation
-    | MutationAnnotation
-    | PluralAnnotation
-    | FilterableAnnotation
-    | FullTextAnnotation
-    | PopulatedByAnnotation
-    | QueryAnnotation
-    | PrivateAnnotation
-    | SelectableAnnotation
-    | SettableAnnotation
-    | TimestampAnnotation
-    | UniqueAnnotation
-    | SubscriptionAnnotation
-    | JWTClaimAnnotation
-    | JWTPayloadAnnotation
-    | RelayIDAnnotation;
-
-export enum AnnotationsKey {
-    authentication = "authentication",
-    authorization = "authorization",
-    coalesce = "coalesce",
-    customResolver = "customResolver",
-    cypher = "cypher",
-    default = "default",
-    filterable = "filterable",
-    fulltext = "fulltext",
-    id = "id",
-    jwtClaim = "jwtClaim",
-    jwtPayload = "jwtPayload",
-    key = "key",
-    limit = "limit",
-    mutation = "mutation",
-    plural = "plural",
-    populatedBy = "populatedBy",
-    private = "private",
-    query = "query",
-    selectable = "selectable",
-    settable = "settable",
-    subscription = "subscription",
-    subscriptionsAuthorization = "subscriptionsAuthorization",
-    timestamp = "timestamp",
-    unique = "unique",
-    relayId = "relayId",
+export interface Annotation {
+    readonly name: string;
 }
```

Comment:
While I wouldn't consider this a blocking problem. I want to mention that a Typescript interface is a bit brittle on this case due to duck-typing.

What I mean is that any object with a property `name` will fulfil this interface (and hence, passed as ok on any function using this type), including classes **not** extending from this interface

By comparison, while the union is, undoubtedly, more verbose, it will statically check for the Annotation being one of the classes defined in it. Making it more robust for type checking